llmfoundry/utils/model_download_utils.py

```diff
@@ -183,53 +190,98 @@ def _recursive_download(
     """
-    formatted_model_name = f'models/{model_name}'.replace('/', '--')
     with requests.Session() as session:
-        session.headers.update({'Authorization': f'Bearer {token}'})
-
-        download_start = time.time()
-
         # Temporarily suppress noisy SSL certificate verification warnings if ignore_cert is set to True
         with warnings.catch_warnings():
             if ignore_cert:
                 warnings.simplefilter('ignore', category=InsecureRequestWarning)
 
-            # Only downloads the blobs in order to avoid downloading model files twice due to the
-            # symlnks in the Hugging Face cache structure:
-            _recursive_download(
-                session,
-                cache_base_url,
-                # Trailing slash to indicate directory
-                f'{formatted_model_name}/blobs/',
-                save_dir,
-                ignore_cert=ignore_cert,
-            )
-        download_duration = time.time() - download_start
-        log.info(
-            f'Downloaded model {model_name} from cache server in {download_duration} seconds'
+            _recursive_download(session,
+                                url,
+                                '',
+                                save_dir,
+                                ignore_cert=ignore_cert)
+
+
+def download_from_oras(model: str,
+                       config_file: str,
+                       credentials_dir: str,
+                       save_dir: str,
+                       concurrency: int = 10):
+    """Download from an OCI-compliant registry using oras.
+
+    Args:
+        model: The name of the model to download.
+        config_file: Path to a YAML config file that maps model names to registry paths.
+        credentials_dir: Path to a directory containing credentials for the registry. It is expected to contain three
+            files: `username`, `password`, and `registry`, each of which contains the corresponding credential.
+        save_dir: Path to the directory where files will be downloaded.
+        concurrency: The number of concurrent downloads to run.
+    """
+    if shutil.which(ORAS_CLI) is None:
+        raise Exception(
+            f'oras cli command `{ORAS_CLI}` is not found. Please install oras: https://oras.land/docs/installation '
         )
+
+    def _read_secrets_file(secret_file_path: str,):
+        try:
+            with open(secret_file_path, encoding='utf-8') as f:
+                return f.read().strip()
+        except Exception as error:
+            raise ValueError(
+                f'secrets file {secret_file_path} failed to be read') from error
+
+    secrets = {}
+    for secret in ['username', 'password', 'registry']:
+        secrets[secret] = _read_secrets_file(
+            os.path.join(credentials_dir, secret))
+
+    with open(config_file, 'r', encoding='utf-8') as f:
+        configs = yaml.safe_load(f.read())
+
+    path = configs[model]
```

Comment:
```suggestion
    path = configs['models'][model]
```

as per our config format changes to include tokenizers as well